src/sparseml/pytorch/torchvision/export_onnx.py

```diff
@@ -0,0 +1,212 @@
+    default=None,
+    help="Optional path to the dataset-specific mapping from "
+    "numeric labels to human-readable class strings. "
+    "Expected to be a path to a .json file containing "
+    "a serialized dictionary",
+)
+@click.option(
+    "--num-samples",
+    type=int,
+    default=-1,
+    help="The number of samples to export along with the model onnx "
+    "and pth files (sample inputs and labels as well as the "
+    "outputs from model execution)",
+)
+@click.option(
+    "--onnx-opset",
+    type=int,
+    default=DEFAULT_ONNX_OPSET,
+    help="The onnx opset to use for exporting the model",
+)
+@click.option(
+    "--save-dir",
+    type=click.Path(dir_okay=True, file_okay=False, path_type=Path),
+    default="torchvision",
+    help="The path to the directory for saving results",
+)
+@click.option(
+    "--convert-qat/--no-convert-qat",
+    default=True,
+    is_flag=True,
+    help="if True, exports of torch QAT graphs will be converted to a fully quantized "
+    "representation. Default is True",
+)
+@click.option(
+    "--interpolation",
+    default="bilinear",
+    type=str,
+    help="the interpolation method (default: bilinear)",
+)
+@click.option(
+    "--img-resize-size",
+    default=256,
+    type=int,
+    help="the resize size used for validation (default: 256)",
+)
+@click.option(
+    "--img-crop-size",
+    default=224,
+    type=int,
+    help="the central crop size used for validation (default: 224)",
+)
+def main(
+    arch_key: str,
+    checkpoint_path: str,
+    dataset_path: Path,
+    labels_to_class_mapping: Optional[Path],
+    num_samples: int,
+    onnx_opset: int,
+    save_dir: Path,
+    convert_qat: bool,
+    interpolation: str,
+    img_resize_size: int,
+    img_crop_size: int,
+):
+    """
+    SparseML torchvision integration for exporting image classification models to
+    onnx along with sample inputs and outputs
+    """
+
+    os.makedirs(save_dir, exist_ok=True)
+
+    dataset = torchvision.datasets.ImageFolder(
+        dataset_path / "val",
+        presets.ClassificationPresetEval(
+            crop_size=img_crop_size,
+            resize_size=img_resize_size,
+            interpolation=InterpolationMode(interpolation),
+        ),
+    )
+    num_classes = len(dataset.classes)
+    data_loader = DataLoader(dataset=dataset, batch_size=1, shuffle=False)
+
+    if arch_key in ModelRegistry.available_keys():
+        model = ModelRegistry.create(key=arch_key, num_classes=num_classes)
+    elif arch_key in torchvision.models.__dict__:
+        # fall back to torchvision
+        model = torchvision.models.__dict__[arch_key](
+            pretrained=False, num_classes=num_classes
+        )
+    else:
+        raise ValueError(
+            f"Unable to find {arch_key} in ModelRegistry or in torchvision.models"
+        )
+    checkpoint = torch.load(checkpoint_path)
+    model.load_state_dict(checkpoint["model"])
```

Comment:
Is this True for all our checkpoints? I think in alot of places we use the key `state_dict` to save the checkpoint, could we do something like checkpoint.get("model") or checkpoint.get("state_dict") 

Along similar lines, it could be that someone starts from a self trained checkpoint, in that case checkpoint will not be a dict but a `torch.nn.Module`, maybe look at the following convenience [function](https://github.com/neuralmagic/sparseml/blob/0892908084caa16a0b7bbd00cc8f9f3468c27b82/src/sparseml/pytorch/utils/model.py#L57) ?

Reply:
Ahh great call out - I had carried over checkpoint["model"] from the torchvision checkpoints. @bfineran should i change the checkpoints to be consistent with save_model's checkpoints and then I can use load_model in this export script

Reply:
Fix in #1158